app/portainer/services/axios.ts

```diff
@@ -47,6 +51,22 @@ export function agentInterceptor(config: AxiosRequestConfig) {
 
 axios.interceptors.request.use(agentInterceptor);
 
+export function redirectInterceptor(error: AxiosResponse) {
+  if (error.status === 307 || error.status === 308) {
+    const redirectReason = error.headers['redirect-reason'];
+    if (redirectReason === 'AdminInitTimeout') {
+      window.location.href = '/#!/init/timeout';
+    }
+    return Promise.reject(error);
+  }
+  return Promise.reject(error);
```

Comment:
since you return this anyway, no need for a double reject, just leave the last one

Reply:
fixed